source/extensions/dynamic_modules/abi_version.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+#ifdef __cplusplus
+namespace Envoy {
+namespace Extensions {
+namespace DynamicModules {
+#endif
+// This is the ABI version calculated as a sha256 hash of the ABI header files. When the ABI
+// changes, this value must change, and the correctness of this value is checked by the test.
+const char* kAbiVersion = "749b1e6bf97309b7d171009700a80e651ac61e35f9770c24a63460d765895a51";
```

Comment:
Obviously this is suboptimal because it means we can't even change comments, but it's a start. We can improve in the future.